src/server.jl

```diff
@@ -29,6 +29,38 @@ end
 ##############################
 # server parameter conversions
 ##############################
+struct Param
+    keylist::Vector{String}
+    value::String
+end
+
+function parse_query_dict(query_dict::Dict{String, String})::Vector{Param}
+    params = Vector{Param}()
+    for (key, value) in query_dict
+        keylist = replace.(split(key, "["), "]"=>"")
+        push!(params, Param(keylist, value))
+    end
+
+    return params
+end
+
+function convert_to_dict(params::Vector{Param})::Dict{String, Any}
+    deserialized_dict = Dict{String, Any}()
+
+    for param in params
+        current = deserialized_dict
+        for part in param.keylist[1:end-1]
+            current = get!(current, part, Dict{String, Any}())
```

Comment:
Change to `get!(current, part) do` format? That would create the `Dict` only if necessary.

Reply:
done